common/src/main/java/org/corfudb/common/util/CompatibilityVectorUtils.java

```diff
@@ -0,0 +1,149 @@
+package org.corfudb.common.util;
+
+import com.google.protobuf.ByteString;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A collection of {@link BitSet}s representing the Corfu features enabled so far.
+ * <p>
+ * Creates the Compatibility Vectors that contain the information about the Corfu features enabled
+ * so far as per the current stage of the code during the compile time of the class.
+ * The vectors are of type {@link ByteString} in which each bit will represent if the corresponding
+ * feature is enabled or not based on its value (0 or 1).
+ * <p>
+ * Client-Server Transmission:
+ * The BitSet values are already converted to the ByteString format and are accessed through
+ * {@link #getCompatibilityVectors()} method. These will be set in the
+ * Protobuf header HeaderMsg.ProtocolVersionMsg value (of both RequestMsg and ResponseMsg),
+ * and sent to the receiver. The receiver will then check whether the specific features are set or
+ * not using {@link CompatibilityVectorUtils#isFeatureEnabled(int, ByteString)} method.
+ * <p>
+ * For adding new Features:
+ * - <b>Introduce</b> them by creating an entry in this class's {@link Feature} enum structure.
+ * <p>
+ * Created by cgudisagar on 5/3/21.
+ */
+@Slf4j
+public class CompatibilityVectorUtils {
+
```

Comment:
Should we add stubs to the places where Compatibility Vector will be used? like have a log of "No feature needs to be specially handled" or something like that. I think we want to be clear about where it will checked at this time.

Reply:
For now, these bitset vectors will be used only in the Handler methods on the server-side and the client-side as they are specific to the features. The only other place was in the handshake handler part but we already have version checks there.
So, in my opinion, the checks can be placed once we have the first feature from the next release.